fdbbackup/backup.actor.cpp

```diff
@@ -2400,6 +2460,74 @@ ACTOR Future<Void> describeBackup(const char *name, std::string destinationConta
 	return Void();
 }
 
+// If restoreVersion is invalidVersion or latestVersion, use the maximum or minimum restorable version respectively for
+// selected key ranges. If restoreTimestamp is specified, any specified restoreVersion will be overriden to the version
+// resolved to that timestamp.
+ACTOR Future<Void> queryBackup(const char* name, std::string destinationContainer,
+                               Standalone<VectorRef<KeyRangeRef>> keyRangesFilter, Version restoreVersion,
+                               std::string originalClusterFile, std::string restoreTimestamp) {
+	// Resolve restoreTimestamp if given
+	if (!restoreTimestamp.empty()) {
+		if (originalClusterFile.empty()) {
+			printf("Error: an original cluster file must be given in order to resolve restore target timestamp '%s'\n",
+			       restoreTimestamp.c_str());
+			return Void();
+		}
+
+		if (!fileExists(originalClusterFile)) {
+			printf("Error: original source database cluster file '%s' does not exist.\n", originalClusterFile.c_str());
+			return Void();
+		}
+
+		Database origDb = Database::createDatabase(originalClusterFile, Database::API_VERSION_LATEST);
+		Version v = wait(timeKeeperVersionFromDatetime(restoreTimestamp, origDb));
+		printf("Timestamp '%s' resolves to version %" PRId64 "\n", restoreTimestamp.c_str(), v);
+		restoreVersion = v;
+	}
+
+	try {
+		state Reference<IBackupContainer> bc = openBackupContainer(name, destinationContainer);
+		if (restoreVersion == invalidVersion) {
+			printf("Using the maximum restorable version for the specified key ranges.\n");
+			BackupDescription desc = wait(bc->describeBackup());
+			if (!desc.maxRestorableVersion.present()) {
+				printf("Error: the specified backup is not restorable to any version.\n");
+				return Void();
+			}
+			restoreVersion = desc.maxRestorableVersion.get();
```

Comment:
It is ok to use the same min and max restorable version for all key ranges, as the first version for this issue.

Different key-ranges' min and max restorable versions can be different and the restorable version range for a small key range should be larger than or equal to the restorable version range for a larger key range. 

We should support that although maybe not in the first version.